include/jemalloc/internal/rtree_inlines.h

```diff
@@ -180,9 +188,9 @@ rtree_elm_acquire(tsdn_t *tsdn, rtree_t *rtree, rtree_ctx_t *rtree_ctx,
 JEMALLOC_INLINE extent_t *
 rtree_elm_read_acquired(tsdn_t *tsdn, const rtree_t *rtree, rtree_elm_t *elm) {
 	extent_t *extent;
-
-	assert(((uintptr_t)elm->pun & (uintptr_t)0x1) == (uintptr_t)0x1);
-	extent = (extent_t *)((uintptr_t)elm->pun & ~((uintptr_t)0x1));
+	void *ptr = atomic_load_p(&elm->child_or_extent, ATOMIC_RELAXED);
```

Comment:
Three questions:
1) What is the performance impact of relaxed atomic loads versus plain memory loads?
2) Does a relaxed atomic read do anything besides guard against torn reads?  I think it causes actual reads (akin to ```volatile```), but I want to be sure my mental model isn't way off.
3) Why do we need to use an atomic load here when the function is specifically used for reading when we've already acquired the elm?  Related to (2), I think it's necessary because there are no explicit dependencies that would otherwise guarantee consistency between atomic and non-atomic accesses, but I'm not sure.